### ISSUE TITLE
missing dependencies in nuclide-debugger-common?

### ISSUE THREAD

**Issue by lloiser** [facebook/nuclide#1502]
I'm currently trying to write a atom-ide-ui debugger.
As mentioned in the [README of atom-ide-ui](https://github.com/facebook-atom/atom-ide-ui/blob/master/README.md)

> Debugger integrations should be modeled after [atom-ide-debugger-node](https://atom.io/packages/atom-ide-debugger-node).

Which uses `nuclide-debugger-common/AutoGenLaunchAttachProvider`.

So I started a new project and added `flow-bin` (0.70.0) and `nuclide-debugger-common` (0.7.0) to my dependencies.

Now `flow status` reports 52 errors when I import `nuclide-debugger-common/AutoGenLaunchAttachProvider`.
Here is the `flow status` output: [errors-52.txt](https://github.com/facebook/nuclide/files/1934950/errors-52.txt)

Once I added https://github.com/facebook/nuclide/blob/master/flow-libs/atom.js.flow to my `flow-libs` folder (and add it in the flowconfig too) the number dropped to 38: [errors-38.txt](https://github.com/facebook/nuclide/files/1934970/errors-38.txt)

After adding 
```
type IDisposable = {
  dispose(): mixed
};
```
to `atom.js.flow` the number of errors dropped to 29: [errors-29.txt](https://github.com/facebook/nuclide/files/1934972/errors-29.txt)

I just want to build a new debugger... 😭 

Here is a minimal project to reproduce it
[my-debugger.zip](https://github.com/facebook/nuclide/files/1934975/my-debugger.zip)

Am I missing something?
Or is the npm package `nuclide-debugger-common` faulty (and maybe others too)?

**Comment by lloiser** [comment]
Also leaving aside these problem, I now have another problem.

As suggested I tried to model my debugger based on the [atom-ide-debugger-node](https://atom.io/packages/atom-ide-debugger-node). It uses the `AutoGenLaunchAttachProvider` to create the config UI and start the debugger.

Unfortunately there is no way to provide my own `adaperExecutable`. Which (I assume) is the actual debug adapter.

https://github.com/facebook/nuclide/blob/5aa45d09ea3caa0cd88215bbdd2e1fbe92c0ab27/modules/nuclide-debugger-common/AutoGenLaunchAttachUiComponent.js#L445

So I cannot use the AutoGen stuff and in the end have to create my own UI and start the debugger myself.

TBH this does not seem right... 

https://github.com/facebook/nuclide/blob/5aa45d09ea3caa0cd88215bbdd2e1fbe92c0ab27/modules/nuclide-debugger-common/AutoGenLaunchAttachUiComponent.js#L441-L452

IMO options like `adapterExecutable`, `capabilities` and `threadsComponentTitle` should be configurable from the outside (and maybe have a meaningful default if none is provided).

**Comment by pelmers** [comment]
@Goom11 sounds reasonable to add a way to provide an adapterExecutable.

it looks like currently the only supported ones are hardcoded in debugger-registry.js

**Comment by vertexclique** [comment]
@pelmers yes, moreover it checks against the existing keys which are not visible from the point of debugger plugin developer. also if there is a way to supply type information to existing adapters like `NATIVE_LLDB` and `NATIVE_GDB` would be cool for other languages. So we can override them and use lldb scripting for example.
